File: src/main/java/xyz/oribuin/lilori/commands/music/CmdStop.java
package xyz.oribuin.lilori.commands.music;

import xyz.oribuin.lilori.managers.command.Command;
import xyz.oribuin.lilori.managers.command.CommandEvent;
import xyz.oribuin.lilori.managers.music.TrackManager;

import java.util.Collections;

public class CmdStop extends Command {

    public CmdStop() {
        this.name = "Stop";
        this.description = "Stops playing Music.";
        this.aliases = Collections.emptyList();
    }

    public void executeCommand(CommandEvent event) {
        TrackManager tm = TrackManager.getInstance(event.getGuild());

        if (!event.getGuild().getAudioManager().isConnected()) {
            event.reply(event.getAuthor() + ", There is no active Audio Track.");
            return;
        }

        tm.getMusicManager().getAudioManager(event.getGuild()).setSendingHandler(null);
        tm.getMusicManager().getAudioManager(event.getGuild()).closeAudioConnection();
        event.reply(event.getAuthor().getAsMention() + ", Ended the audio track. :wave:");
    }
}

File: src/main/kotlin/xyz/oribuin/lilori/commands/music/CmdStop.kt
package xyz.oribuin.lilori.commands.music

import net.dv8tion.jda.api.EmbedBuilder
import xyz.oribuin.lilori.LilOri
import xyz.oribuin.lilori.handler.Category
import xyz.oribuin.lilori.handler.Command
import xyz.oribuin.lilori.handler.CommandEvent
import xyz.oribuin.lilori.managers.music.TrackManager.Companion.getInstance
import java.awt.Color

class CmdStop(bot: LilOri) : Command(bot) {
    init {
        name = "Stop"
        category = Category(Category.Type.MUSIC)
        description = "Stops playing Music."
        aliases = emptyList()
        arguments = emptyList()
    }

    override fun executeCommand(event: CommandEvent) {

        val tm = getInstance(event.guild)
        (tm ?: return)

        if (!event.guild.audioManager.isConnected) {
            event.reply(event.author.toString() + ", There is no active Audio Track.")
            return
        }

        tm.playerManager.shutdown()
        tm.musicManager.getAudioManager(event.guild).sendingHandler = null
        tm.musicManager.getAudioManager(event.guild).closeAudioConnection()
        val embedBuilder = EmbedBuilder()
                .setAuthor("\uD83C\uDFB5 Stopping music")
                .setColor(Color.RED)
                .setDescription("Successfully stopped playing music, Leaving Voice Channel!")
                .setFooter("Created by Oribuin", "https://imgur.com/ssJcsZg.png")

        event.channel.sendMessage(event.author.asMention).embed(embedBuilder.build()).queue()
    }
}